src/main/java/com/enderio/base/common/util/ExperienceUtil.java

```diff
@@ -1,13 +1,61 @@
 package com.enderio.base.common.util;
 
-import com.enderio.api.misc.Vector2i;
 import net.minecraft.world.entity.player.Player;
 
 public class ExperienceUtil {
 
     // 1 exp = 20 mb
     public static int EXP_TO_FLUID = 20;
 
+    /**
+     * Vanilla way of calculating experience points required for level up.
+     * @param currentLevel - the current level of player. The level up cost depends on currentLevel and not on the level you are trying to reach
+     * @return experience - experience cost to level up
+     */
+
+    public static int getXpNeededForNextLevel(int currentLevel) {
+        if (currentLevel >= 30) {
+            return 112 + (currentLevel - 30) * 9;
+        } else {
+            return currentLevel >= 15 ? 37 + (currentLevel - 15) * 5 : 7 + currentLevel * 2;
+        }
+    }
+
+    /**
+     * Returns the total xp required to reach the target level ( from 0 )
+     * @param level - level to reach
+     * @return experience - total xp cost
+     */
+    public static int getTotalXpFromLevel(int level) {
```

Comment:
any chance to make this long? and accept long in `getTotalLevelFromXp`, as usually xp is just used as levels + the xp, so this method breaks faster than actual levels you can get, and I still think that for heavy modded moddpack we really need to start going away from ints, so many issues in so many mods as people just keep going higher. 
This breaks at 21863+ level, that is very high, but i feel like its doable with how many weird mods this can be combined with

Reply:
Yeah that is the plan. The only issue `Player#giveExperiencePoints` takes int input. I will replace rest of the values with longs

Reply:
Another issue that fluidStacks used to store XP use integers as well.

Reply:
using long for this would allow to easier future proof to hopefully improve containers later too, but for now would at least give easy way to check if XP will overflow and limit operations to only fill containers up to int max and not worry about it overflowing when calculating

Reply:
Any suggestions on how I should do that ?

Reply:
Looks good! I will convert them over :)